phpunit/functional/ReservationTest.php

```diff
@@ -176,4 +186,727 @@ public function testDeleteRecurrentReservation(): void
+    public function testCanPurgeWithReserveanitemRight(): void
+    {
+        $this->login();
+
+        // Test with PURGE right
+        $_SESSION['glpiactiveprofile']['reservation'] = PURGE;
+        $this->assertTrue((bool) \Reservation::canPurge());
+
+        // Test with RESERVEANITEM right (simplified interface case)
+        $_SESSION['glpiactiveprofile']['reservation'] = \ReservationItem::RESERVEANITEM;
+        $this->assertTrue((bool) \Reservation::canPurge());
+
+        // Test with both rights
+        $_SESSION['glpiactiveprofile']['reservation'] = PURGE | \ReservationItem::RESERVEANITEM;
+        $this->assertTrue((bool) \Reservation::canPurge());
+
+        // Test with unrelated right
+        $_SESSION['glpiactiveprofile']['reservation'] = READ;
+        $this->assertFalse((bool) \Reservation::canPurge());
+
+        // Test with no rights
+        $_SESSION['glpiactiveprofile']['reservation'] = 0;
+        $this->assertFalse((bool) \Reservation::canPurge());
+    }
+
+    /**
+     * Test that canCreate and canDelete methods work with RESERVEANITEM right
+     */
+    public function testCanCreateAndDeleteWithReserveanitemRight(): void
+    {
+        $this->login();
+
+        // Test canCreate with RESERVEANITEM right
+        $_SESSION['glpiactiveprofile']['reservation'] = \ReservationItem::RESERVEANITEM;
+        $this->assertTrue((bool) \Reservation::canCreate(), "canCreate should return truthy value with RESERVEANITEM right");
+
+        // Test canDelete with RESERVEANITEM right - canDelete only checks RESERVEANITEM
+        $this->assertTrue((bool) \Reservation::canDelete(), "canDelete should return truthy value with RESERVEANITEM right");
+
+        // Test with no rights
+        $_SESSION['glpiactiveprofile']['reservation'] = 0;
+        $this->assertFalse((bool) \Reservation::canCreate());
+        $this->assertFalse((bool) \Reservation::canDelete());
+    }
+
+    /**
+     * Test canChildItem method for reservation ownership
+     */
+    public function testCanChildItemOwnership(): void
+    {
+        $this->login();
+
+        // Create a computer and reservation item
+        $computer = $this->createItem("Computer", [
+            "name" => "test computer",
+            "entities_id" => 0,
+        ]);
+        $res_item = $this->createItem("ReservationItem", [
+            "itemtype" => "Computer",
+            "items_id" => $computer->getID(),
+            "is_active" => true,
+            "entities_id" => 0,
+        ]);
+
+        // Create a reservation owned by current user
+        $reservation = new \Reservation();
+        $reservation_id = $reservation->add([
+            'begin' => '2024-01-01 10:00:00',
+            'end' => '2024-01-01 12:00:00',
+            'reservationitems_id' => $res_item->getID(),
+            'users_id' => $_SESSION['glpiID'],
+        ]);
+        $this->assertGreaterThan(0, $reservation_id);
+        $this->assertTrue($reservation->getFromDB($reservation_id));
```

Comment:
Please use `createItem()` whenever possible; it is more precise. `This also applies to the other cases below`.
```suggestion
        $reservation = $this->createItem(
            'Reservation',
            [
                'begin' => '2024-01-01 10:00:00',
                'end' => '2024-01-01 12:00:00',
                'reservationitems_id' => $res_item->getID(),
                'users_id' => $_SESSION['glpiID'],
            ]
        );
```